homeassistant/components/sms/sensor.py

```diff
@@ -14,48 +14,41 @@
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the GSM Signal Sensor sensor."""
     gateway = hass.data[DOMAIN][SMS_GATEWAY]
-    entities = []
     imei = await gateway.get_imei_async()
-    name = f"gsm_signal_imei_{imei}"
-    entities.append(
-        GSMSignalSensor(
-            hass,
-            gateway,
-            name,
-        )
+    async_add_entities(
+        [
+            GSMSignalSensor(
+                hass,
+                gateway,
+                imei,
+                SensorEntityDescription(
+                    key="signal",
+                    name="GSM Signal IMEI",
+                    device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
+                    unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+                    entity_registry_enabled_default=False,
+                ),
+            )
+        ],
+        True,
     )
-    async_add_entities(entities, True)
 
 
 class GSMSignalSensor(SensorEntity):
     """Implementation of a GSM Signal sensor."""
 
-    def __init__(
-        self,
-        hass,
-        gateway,
-        name,
-    ):
+    def __init__(self, hass, gateway, imei, description):
         """Initialize the GSM Signal sensor."""
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, imei)},
+            "name": "SMS Gateway",
+        }
+        self._attr_name = f"{description.name} {imei}"
+        self._attr_unique_id = imei
```

Comment:
```suggestion
        self._attr_unique_id = str(imei)
```
This seems to be a number